include/runtime/header.h

```diff
@@ -5,49 +5,63 @@
 #include <gmp.h>
 #include <mpfr.h>
 
+// the actual length is equal to the block header with the gc bits masked out.
+#define len(s) ((s)->h.hdr & 0xffff3fffffffffff)
```

Comment:
A macro for masking out the GC bits sounds like a good thing to have, but a macro called `len` should just mask with `0x3fffffffffff` rather than ever returning parts of the high bytes. I also doubt that the immediate in the code can be shared, and while it might not actually be more expensive I doubt this immediate could possibly be cheaper to construct than `0x3fffffffffff`.

Reply:
Plus, distinguishing a `len` and a macro to `clear_gc_bits` now will make it easier in the future if we want to try changing representations in a way that makes a difference between between suppressing GC state and computing string length.